b2sdk/utils.py

```diff
@@ -211,6 +212,54 @@ def is_file_readable(local_path, reporter=None):
     return True
 
 
+def get_file_mtime(local_path, rounded=True):
+    """
+    Get modification time of a file in milliseconds.
+
+    :param local_path: a file path
+    :type local_path: str
+    :param rounded: floating point time should be rounded instead of stripped
+    :type rounded: bool
+    :rtype: int
+    """
+    mod_time = os.path.getmtime(local_path) * 1000
+    if rounded:
+        # Using round() is non backward-compatible change, as some files
+        # which were uploaded with older version would have newer mtime
+        # on the destination, thus b2 may fail.
+        # The default will be changed in v2. See #617 for details.
+        mod_time = round(mod_time)
+
+    return int(mod_time)
+
+
+def set_file_mtime(local_path, mod_time_millis, rounded=True):
+    """
+    Set modification time of a file in milliseconds.
+
+    :param local_path: a file path
+    :type local_path: str
+    :param mod_time_millis: time to be set
+    :type mod_time_millis: int
+    :param rounded: if False, the time is updated with small delta
+    :type rounded: bool
+    """
+    mod_time = mod_time_millis / 1000.0
+    if not rounded:
+        # We have to convert it this way to avoid differences when mtime
```

Comment:
Hi. The problem is not the filesystem nor the resolution. It's just because of floating-point arithmetic as I pointed out in the comment. Here is the explanation - https://docs.python.org/3/tutorial/floatingpoint.html

If your OS is storing mtime as a float and B2 as an integer, and current logic doesn't use rounding when converting (b2sdk doesn't), I can take one particular value, convert it forth and back and I will end up with different value.
See this simple snippet (which doesn't touch filesystem btw)
```python
# b2 file mtime, fetched from the cloud
downloaded_b2_mtime = 1093258377393

# convert it in a way b2sdk does, to save it to the disk in a first download
local_mtime = downloaded_b2_mtime / 1000.0

# convert it back, to compare in case of second download
calculated_b2_mtime = int(local_mtime * 1000)

print(downloaded_b2_mtime, calculated_b2_mtime, downloaded_b2_mtime == calculated_b2_mtime)
```
```
1093258377393 1093258377392 False
```
If you have a file with 1093258377393 it will be downloaded over and over again.

If you add rounding - `int(round(local_mtime * 1000))` - you will never have different values, no matter what value you have used as an input. Simply add rounding would be the best solution, but the problem is that in some cases when backing up from local to B2 cloud, mtime of some files can be calculated with the new (correct!) algorithm as e.g. 1093258377392 so the CLI will fail, complaining that the destination file is newer. 

My solution makes sure that when we read mtime in the next round it will be converted to 1093258377393, however 1093258377392 should be correct value, but on the cloud we have 1093258377393 already.

I ran that kind of test with B2 CLI and real file, and the above problem is simply to reproduce.